clouddrift/wavelet.py

```diff
@@ -0,0 +1,518 @@
+        )
+    else:
+        _wavefft = _wavefft * np.exp(1j * -om * (time_length_ + 1) / 2)
+
+    # here we should be able to automate the tiling without assuming extra dimensions of wave
+    X_ = np.tile(
+        np.expand_dims(np.fft.fft(x_), (-3, -2)),
+        (1, order_length, freq_length, 1),
+    )
+
+    # finally the transform; return precision of input `x``; central part only
+    complex_dtype = np.cdouble if x.dtype == np.single else np.csingle
+    wtx = np.fft.ifft(X_ * np.conj(_wavefft)).astype(complex_dtype)
+    wtx = wtx[..., index]
+    # remove extra dimensions
+    wtx = np.squeeze(wtx)
+
+    return wtx
+
+
+def morsewave(
+    n: int,
+    ga: float,
+    be: float,
+    rad_freq: np.ndarray,
+    order: Optional[int] = 1,
+    norm: Optional[str] = "bandpass",
+) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    Generalized Morse wavelets of Olhede and Walden (2002).
+
+    Parameters
+    ----------
+    n: int
+       Length of the wavelet.
+    ga: float
+       Gamma parameter of the wavelet.
+    be: float
+       Beta parameter of the wavelet.
+    rad_freq: np.ndarray
+       The radian frequencies at which the Fourier transform of the wavelets
+       reach their maximum amplitudes. rad_freq is between 0 and 2 * np.pi * 0.5,
+       the normalized Nyquist radian frequency.
+    order: int
+        Wavelet order, default is 1.
+    norm:  str, optional
+       Normalization for the wavelets. Default is "bandpass".
+       "bandpass" uses "bandpass normalization", meaning that the FFT of the wavelet
+       has a peak value of 2 for all frequencies rad_freq. "energy" uses the unit
+       energy normalization. The time-domain wavelet energy np.sum(np.abs(wave)**2)
+       is then always unity.
+
+    Returns
+    -------
+    wave : np.ndarray
+        Time-domain wavelets. wave will be of shape (n,np.size(rad_freq),k).
+    wavefft: np.ndarray
+        Frequency-domain wavelets. wavefft will be of shape (n,np.size(rad_freq),k).
+
+    Examples
+    --------
+    To write.
+
+    See Also
+    --------
+    :func:`wavetrans`, `morsefreq`
+    """
+    # add a test for rad_freq being a numpy array
+    # initialization
+    wave = np.zeros((n, order, len(rad_freq)), dtype=np.cdouble)
+    wavefft = np.zeros((n, order, len(rad_freq)), dtype=np.cdouble)
+
+    # call to morsewave take only ga and be as float, no array
+    fo, _, _ = morsefreq(ga, be)
+    for i in range(len(rad_freq)):
+        wave_tmp = np.zeros((n, order), dtype=np.cdouble)
+        wavefft_tmp = np.zeros((n, order), dtype=np.cdouble)
+
+        # wavelet frequencies
+        fact = np.abs(rad_freq[i]) / fo
+        # norm_rad_freq first dim is n points
+        norm_rad_freq = 2 * np.pi * np.linspace(0, 1 - 1 / n, n) / fact
+        if norm == "energy":
+            if be == 0:
+                wavezero = np.exp(-(norm_rad_freq**ga))
+            else:
+                wavezero = np.exp(be * np.log(norm_rad_freq) - norm_rad_freq**ga)
```

Comment:
Could this not be just

```suggestion    
            wavezero = np.exp(be * np.log(norm_rad_freq) - norm_rad_freq**ga)
```

if `be == 0`, the first term in the exponent argument drops.